scripts/react-native-xcode.sh

```diff
@@ -145,11 +146,6 @@ case "$PLATFORM_NAME" in
     ;;
 esac
 
-USE_HERMES=
-if [[ "$BUNDLE_PLATFORM" == "macos" && -f "$HERMES_PATH" ]]; then
-  USE_HERMES=true
-fi
```

Comment:
Looks like the approach here was: "If platform is Mac and Hermes exists" which assumed Hermes depedency is optional and installed on per project base (which is aligned with your previous comment @alloy).

Did a little update - USE_HERMES can be (and should) be set by a user explicitly. No platform checks.

Reply:
> Did a little update - USE_HERMES can be (and should) be set by a user explicitly.

Why have another extra option? Having the user install the engine is already an explicit choice. It’s also not clear to me in this PR _where/how_ the user should set that variable.

Reply:
> Having the user install the engine is already an explicit choice.

In React Native itself, the engine is a dependency and will be shipped just like an Android one. Users will not have to install it, just enable it by turning the flag.

I feel like relying on its presence inside `node_modules` might be a bit confusing, say - it's still in `node_modules` while somebody finished experimenting with Hermes and wanted to get back to regular setup. 

While this pattern generally makes sense to me, I don't think it's what RN users would expect. On contrary, we already use flags like that one throughout the app.

Reply:
You would set it inside `Bundle JS` script phase, where you set other variables that you want to pass to `react-native-xcode.sh`.

Reply:
Ideally, we would do something like `#if __has_include(<hermes/hermes.h>)` to detect if Hermes is actually included. Unfortunately, I am not sure whether we can detect that from bash script, that's why I ended up using USE_HERMES flag.

Reply:
> I don't think it's what RN users would expect. On contrary, we already use flags like that one throughout the app.

> You would set it inside `Bundle JS` script phase, where you set other variables that you want to pass to `react-native-xcode.sh`.

Ok, consistency is worth something 👍

Reply:
Another reason is if people want to ship two versions of their app to experiment with, one running on Hermes, and one without. Without a flag they'd have to change their package.json to build the two different variants.